graphql/resolve/add_mutation_test.yaml

```diff
@@ -1100,6 +1100,498 @@
+            ],
+            "readHours": "4d2hr"
+        }
+    }
+  dgquery: |-
+    query {
+      LibraryMember_1(func: eq(Member.name, "Alice")) @filter(type(LibraryMember)) {
+        uid
+      }
+      LibraryMember_2(func: eq(Member.refID, "101")) @filter(type(LibraryMember)) {
+        uid
+      }
+      LibraryMember_3(func: eq(Member.refID, "101")) @filter(type(Member)) {
+        uid
+      }
+    }
+  qnametouid: |-
+    {
+      "LibraryMember_3": "0x11"
+    }
+  error2:
+    {
+      "message": "failed to rewrite mutation payload because interface Member; field refID: id 101 already exists
+       for one of the implementing type of interface"
+    }
+
+-
+  name: "Add mutation on implementation type which have inherited @id field with unique argument -3"
+  explanation: "Node with refID:101 already exist in same mutated type, returns error "
+  gqlmutation: |
+    mutation addLibraryMember($input: AddLibraryMemberInput!) {
+      addLibraryMember(input: [$input], upsert: false) {
+        libraryMember {
+          refID
+        }
+      }
+    }
+  gqlvariables: |
+    {
+        "input": {
+            "refID": "101",
+            "name": "Alice",
+            "itemsIssued": [
+                "Intro to Go",
+                "Parallel Programming"
+            ],
+            "readHours": "4d2hr"
+        }
+    }
+  dgquery: |-
+    query {
+      LibraryMember_1(func: eq(Member.name, "Alice")) @filter(type(LibraryMember)) {
+        uid
+      }
+      LibraryMember_2(func: eq(Member.refID, "101")) @filter(type(LibraryMember)) {
+        uid
+      }
+      LibraryMember_3(func: eq(Member.refID, "101")) @filter(type(Member)) {
+        uid
+      }
+    }
+  qnametouid: |-
+    {
+      "LibraryMember_2": "0x11"
+    }
+  error2:
+    {
+      "message": "failed to rewrite mutation payload because Type LibraryMember; field refID: id 101 already exists"
+    }
+
+-
+  name: "Add upsert mutation on implementation type which have inherited @id field with unique argument -1"
+  explanation: "node with @id field doesn't exist in any of the implementing type, we will add the node"
+  gqlmutation: |
+    mutation addLibraryMember($input: AddLibraryMemberInput!) {
+      addLibraryMember(input: [$input], upsert: true) {
+        libraryMember {
+          refID
+        }
+      }
+    }
+  gqlvariables: |
+    {
+        "input": {
+            "refID": "101",
+            "name": "Alice",
+            "itemsIssued": [
+                "Intro to Go",
+                "Parallel Programming"
+            ],
+            "readHours": "4d2hr"
+        }
+    }
+  dgquery: |-
+    query {
+      LibraryMember_1(func: eq(Member.name, "Alice")) @filter(type(LibraryMember)) {
+        uid
+      }
+      LibraryMember_2(func: eq(Member.refID, "101")) @filter(type(LibraryMember)) {
+        uid
+      }
+      LibraryMember_3(func: eq(Member.refID, "101")) @filter(type(Member)) {
+        uid
+      }
+    }
+  dgmutations:
```

Comment:
No uid is returned in the result of existence query. The upsert is not taking place. Can you add qnametouid map to ensure that the upsert actually takes place.